server/database/dbOperations.ts

```diff
@@ -229,3 +239,397 @@ export const removeTableFromConfig = async (
     throw error;
   }
 };
+
+// Annotated Collections Operations
+
+export const createAnnotatedCollection = async (
+  db: DatabaseConnection,
+  collection: Omit<AnnotatedCollection, "id" | "created_at" | "updated_at">,
+  incidentData?: Omit<Incident, "collection_id">,
+  entries?: Array<{
+    source_table: string;
+    source_id: string;
+    notes?: string;
+  }>,
+): Promise<AnnotatedCollection> => {
+  const pgClient = db as Client;
+
+  try {
+    await pgClient.query("BEGIN");
+
+    // Create the main collection
+    const collectionQuery = `
+      INSERT INTO annotated_collections (name, description, collection_type, status, created_by, is_active, metadata)
+      VALUES ($1, $2, $3, $4, $5, $6, $7)
+      RETURNING *
+    `;
+
+    const collectionResult = await pgClient.query(collectionQuery, [
+      collection.name,
+      collection.description,
+      collection.collection_type,
+      collection.status,
+      collection.created_by,
+      collection.is_active,
+      JSON.stringify(collection.metadata),
+    ]);
+
+    const newCollection = collectionResult.rows[0];
+
+    // Create incident-specific data if provided
+    if (incidentData && collection.collection_type === "incident") {
+      const incidentQuery = `
+        INSERT INTO incidents (collection_id, incident_type, responsible_party, impact_description, supporting_evidence)
+        VALUES ($1, $2, $3, $4, $5)
+      `;
+
+      await pgClient.query(incidentQuery, [
+        newCollection.id,
+        incidentData.incident_type,
+        incidentData.responsible_party,
+        incidentData.impact_description,
+        JSON.stringify(incidentData.supporting_evidence || {}),
+      ]);
+    }
+
+    // Add entries if provided
```

Comment:
Can you explain a little bit more what this means? What are "entries" in this context? I think it is "collection-specific data" but it would be good to clarify in the comments. I think that for this function, more description overall benefits readability.